src/library-authoring/create-library/LibraryCreatePage.jsx

```diff
@@ -0,0 +1,357 @@
+  }
+
+  closeModal = () => {
+    this.setState({ isOpenModal: false });
+  }
+
+  handleClickBreadcrumbs = (event) => {
+    if (!this.state.allowLeave) {
+      event.preventDefault();
+      const pathname = event.target.getAttribute('href');
+      this.openModal({ pathname });
+    }
+  }
+
+  handleBlockedNavigation = (nextLocation) => {
+    const { confirmedNavigation, allowLeave } = this.state;
+    if (!confirmedNavigation && !allowLeave) {
+      this.openModal(nextLocation);
+      return false;
+    }
+
+    return true;
+  }
+
+  handleConfirmNavigationClick = () => {
+    const { lastLocation } = this.state;
+    const { config } = this.context;
+
+    this.closeModal();
+
+    if (lastLocation) {
+      this.setState({
+        confirmedNavigation: true,
+      }, () => {
+        if (lastLocation.pathname === config.STUDIO_BASE_URL) {
+          window.location.href = lastLocation.pathname;
+        } else {
+          this.props.history.push(lastLocation.pathname);
+        }
+      });
+    }
+  }
+
+  render() {
+    const { intl, errorMessage, orgs } = this.props;
+    const { data, isOpenModal, allowLeave } = this.state;
+    const { config } = this.context;
+    const errorTitle = !errorMessage && this.props.errorFields
+      && intl.formatMessage(messages['library.form.generic.error.title']);
+    const errorDescription = errorMessage || this.props.errorFields
+      ? intl.formatMessage(messages['library.form.generic.error.description']) : '';
+
+    return (
+      <div className="container-fluid">
+        <div className="library-create-wrapper">
+          {errorDescription && (
+            <Alert
+              icon={Info}
+              variant="danger"
+              className="form-create-alert"
+            >
+              {errorTitle && <Alert.Heading>{errorTitle}</Alert.Heading>}
+              <p>{truncateMessage(errorDescription)}</p>
+            </Alert>
+          )}
+          <Breadcrumb
+            activeLabel={intl.formatMessage(messages['library.form.breadcrumbs.current'])}
+            clickHandler={this.handleClickBreadcrumbs}
+            links={[
+              { label: intl.formatMessage(messages['library.form.breadcrumbs.home']), url: config.STUDIO_BASE_URL },
+              { label: intl.formatMessage(messages['library.form.breadcrumbs.list']), url: ROUTES.List.HOME },
+            ]}
+          />
+          <div className="wrapper-mast wrapper">
+            <header className="has-actions">
+              <h2 className="page-header h2">{intl.formatMessage(messages['library.form.create.library'])}</h2>
+            </header>
+          </div>
+          <div className="wrapper-content wrapper">
+            <section className="content">
+              <form onSubmit={this.onSubmit} className="form-create">
+                <fieldset>
+                  <ol className="list-input">
+                    <li className="field">
+                      <FormGroup
```

Comment:
I am a little confused as to why your `FormGroup` custom component only includes one form control. In its normal uses, a form group controls multiple form.control or other input components.

Reply:
Yeah, that makes sense as to why it was used, and I guess it is in fact a wrapper for a formgroup. Sometimes a "better name" is more trouble than it is worth.

Reply:
Yes, you are definitely right, it's just a wrapper for a FormGroup component.
Is it better to rename it like FormGroupWrapper or something like that?

Reply:
I wouldn't worry too much about it.